src/ucp/rma/rma_sw.c

```diff
@@ -72,14 +72,19 @@ static ucs_status_t ucp_rma_sw_progress_get(uct_pending_req_t *self)
     ssize_t packed_len = 0;
     ucs_status_t status;
 
-    req->send.lane = ucp_ep_get_am_lane(req->send.ep);
-    status         = ucp_rma_sw_do_am_bcopy(req, UCP_AM_ID_GET_REQ,
-                                            req->send.lane,
-                                            ucp_rma_sw_get_req_pack_cb, req,
-                                            &packed_len);
-    if ((status != UCS_OK) && (status != UCS_ERR_NO_RESOURCE)) {
-        /* completed with error */
-        ucp_request_complete_send(req, status);
+    req->send.lane        = ucp_ep_get_am_lane(req->send.ep);
+    req->send.rma.sreq_id = ucp_send_request_get_id(req);
```

Comment:
why not adding in pack cb?
or add if operations completes ok?

Reply:
inside the pack_cb we don't actually know if AM Bcopy completes with success or not.
so, we do it prior to calling AM Bcopy in order to makes the sreq_id initialized for further usage